tools/src/main/java/org/infinispan/tools/store/migrator/jdbc/AbstractJdbcEntryIterator.java

```diff
@@ -35,7 +36,12 @@ abstract class AbstractJdbcEntryIterator implements Iterator<MarshallableEntry>,
 
       try {
          conn = connectionFactory.getConnection();
-         numberOfRows = tableManager.size(conn);
+         String sizeSql = ((AbstractTableManager<?,?>) tableManager).getSizeSql();
+         ps = conn.prepareStatement(sizeSql);
+         ps.setLong(1, System.currentTimeMillis());
+         rs = ps.executeQuery();
+         rs.next();
+         numberOfRows = rs.getInt(1);
```

Comment:
We need to close the `ps` and `rs` variables.

Reply:
They're already closed by `#close` in normal circumstances. I have added an explicit call to `#close` in the event of an exception.

Reply:
No, they aren't closed in close, because they are reassigned in the lines below.

Reply:
Argh of course :man_facepalming:, updated!